### PR TITLE
Download artifacts to relative locations inside the task directory

### PR DESCRIPTION
**Pull request by dadgar**
This PR:
* Allows for downloading of artifacts to relative locations in the task's directory
* Adds binary path lookup
* Updates JSON documentation
* Makes the default download location the task's local directory.

### CODE REVIEW

File: client/driver/executor/executor.go [hunk 1]
```diff
@@ -159,20 +160,36 @@ func (e *UniversalExecutor) LaunchCmd(command *ExecCommand, ctx *ExecutorContext
...
 
-	// starting the process
+	// Determine the path to run as it may have to be relative to the chroot.
+	path := absPath
+	if e.command.FSIsolation {
+		rel, err := filepath.Rel(e.taskDir, absPath)
+		if err != nil {
+			return nil, err
+		}
+		path = rel
```
Comment (armon) [inline client/driver/executor/executor.go]: Defense in depth, I would re-verify that taskDir is a parent of absPath

File: client/getter/getter.go [hunk 2]
```diff
@@ -59,16 +60,17 @@ func getGetterUrl(artifact *structs.TaskArtifact) (string, error) {
...
+func GetArtifact(artifact *structs.TaskArtifact, taskDir string, logger *log.Logger) error {
 	url, err := getGetterUrl(artifact)
 	if err != nil {
 		return err
 	}
 
 	// Download the artifact
-	if err := getClient(url, destDir).Get(); err != nil {
-		return err
+	dest := filepath.Join(taskDir, artifact.RelativeDest)
```
Comment (armon) [inline client/getter/getter.go]: I would guard this destination as well.

**Comment by armon** [comment]
Minor comments, LGTM

File: client/driver/executor/executor.go (hunk 1 above)
Comment (dadgar) [inline client/driver/executor/executor.go]: Doing the check here is unnecessary as if the path has escaped, it will be invalid inside the chroot.

**Pull request merged**